### PR TITLE
Update 3 Casks: change to license caveat DSL

### PR DESCRIPTION
**Pull request by commitay**
Need to wait for a new tag before we merge this.

### CODE REVIEW

**Comment by vitorgalvao** [comment]
> using interpolation in the caveat DSL doesn't seem to work.

Pretty sure it does, but the syntax is slightly different. Pinging @reitermarkus.

**Comment by vitorgalvao** [comment]
There are working fine with interpolation for me.

File: Casks/segger-jlink.rb [hunk 1]
```diff
@@ -15,9 +15,7 @@
   uninstall pkgutil: 'com.segger.pkg.JLink'
 
   caveats <<~EOS
-    Installing this Cask means you have AGREED to the
-    Segger Downloads License agreement available at
-
+    Installing #{token} means you have AGREED to the license at
       #{url}
```
Comment (vitorgalvao) [inline Casks/segger-jlink.rb]: This doesn’t make much sense because the `url` is a `pkg`.

Comment (commitay) [reply Casks/segger-jlink.rb]: It's weird but it is correct. I couldn't find a URL for the license that wan't the download URL.

**Comment by commitay** [comment]
> These are working fine with interpolation for me.

I don't know what I was doing wrong but they do seem to work.

I'll update and add this to the other PR.

**Comment by commitay** [comment]
Nope, spoke to soon.

@vitorgalvao Can you post an example of how these work with the license DSL?

**Comment by vitorgalvao** [comment]
> Can you post an example of how these work with the license DSL?

The edits should work.

It’s explained in [the docs](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/caveats.md). For [strings](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/caveats.md#caveats-as-a-string) (i.e. when we use the heredoc), we have a few predefined methods for interpolation. For [blocks](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/caveats.md#caveats-as-a-block) (`caveats do`) we can call `#{@cask.whatever}`.

**Pull request merged**